packages/webchat/src/main.tsx

```diff
@@ -88,19 +95,30 @@ class Web extends React.Component<MainProps> {
   async load() {
     this.config = this.extractConfig()
 
+    // TODO: 6 - this implies iframe as well. Do we need this? Why?
     if (this.config.exposeStore) {
       const storePath = this.config.chatId ? `${this.config.chatId}.webchat_store` : 'webchat_store'
       set(window.parent, storePath, this.props.store)
     }
 
-    this.config.overrides && this.loadOverrides(this.config.overrides)
+    if (this.config.overrides) {
+      this.loadOverrides(this.config.overrides)
+    }
 
-    this.config.containerWidth && this.postMessageToParent('setWidth', this.config.containerWidth)
+    // TODO: 7 - again with the iframe stuff. Why do we need this.
+    if (this.config.containerWidth) {
+      this.postMessageToParent('setWidth', this.config.containerWidth)
+    }
 
-    this.config.reference && this.props.setReference!()
+    // TODO: 8 - what is reference?
+    if (this.config.reference) {
+      await this.props.setReference!()
+    }
 
+    // TODO: 9 - don't think we need to fetch bot info from a backend. All this can be set at the frontend level
     await this.props.fetchBotInfo!()
 
+    // TODO: 10 - That's something we should still have as configurable. We just don't start the socket connection right away
     if (!this.isLazySocket()) {
       await this.initializeSocket()
```

Comment:
Yeah, because we want to initialize the webchat once the user clicks on the webchat icon on the bottom right corner. It's useless to initialize before an interaction because it just causes a ton of usage